File: Java-version/LeetCode/src/com/jooc/GazeOffer/Solution_50.java
package com.jooc.GazeOffer;

import com.jooc.Solution;
import com.sun.org.apache.xpath.internal.operations.Bool;

import java.util.HashMap;
import java.util.LinkedHashMap;
import java.util.Map;

public class Solution_50 implements Solution {

    @Override
    public void main() {
        System.out.println(firstUniqChar("leetcode"));
    }

    public char firstUniqChar_0(String s) {
        HashMap<Character, Boolean> map = new HashMap<>();

        for(char c: s.toCharArray()){
            if(map.containsKey(c)){
                map.put(c, false);
            }else{
                map.put(c, true);
            }
        }

        for(char c: s.toCharArray()){
            if(map.get(c)){
                return c;
            }
        }
        return ' ';
    }

    public char firstUniqChar(String s){
        HashMap<Character, Boolean> map = new LinkedHashMap<>();

        char[] chars = s.toCharArray();
        for(char c: chars){
            map.put(c, !map.containsKey(c));
        }
        for(Map.Entry<Character, Boolean> entry: map.entrySet()){
            if(entry.getValue())
                return entry.getKey();
        }

        return ' ';
    }

    public char firstUniqueChar(String s){
        HashMap<>
    }
}

File: Swift-version/LeetCode/LeetCode/Solutions/GazeOffer/Solution_50.swift
//
//  Solution_50.swift
//  LeetCode
//
//  Created by 齐旭晨 on 2021/4/9.
//

import Foundation

class Solution_50: Solution{
    func main() {
        print(firstUniqChar("leetcode"))
    }
    
    func firstUniqChar(_ s: String) -> Character {
        var counter:[Character: Bool] = [:]
        
        for ch in s{
            if counter[ch] == nil{
                counter[ch] = true
            }else{
                counter[ch] = false
            }
        }
        for ch in s{
            if counter[ch]! {
                return ch
            }
        }
        return " "
    }
}
